### PR TITLE
Add function to compute the Jacobian derivative of additional frames rigidly attached to links

### LINKED ISSUE #208: Add frame Jacobian derivative
For the calculation of the acceleration of a frame, it would be useful if Jaxsim could calculate the Jacobian derivative for frames, as it already does for [links](https://github.com/ami-iit/jaxsim/blob/ba9b8da97f72fd51ddc2a1c771991f89c9a8a9a4/src/jaxsim/api/link.py#L389).

### PR DESCRIPTION
**Pull request by xela-95**
Closes #208 

----
📚 Documentation preview 📚: https://jaxsim--210.org.readthedocs.build//210/

### CODE REVIEW

**Comment by xela-95** [comment]
In https://github.com/ami-iit/jaxsim/pull/210/commits/b747a26246927128531f7062533d2c809c3aafd0 I commented out the test against iDynTree and tried to test directly $\dot J$ against the same quantity computing through AD. It seems that the inertial and body cases pass for all models, while the mixed case no.

**Comment by xela-95** [comment]
In https://github.com/ami-iit/jaxsim/pull/210/commits/78b690d2bbfd3a046e025bd69dd6e55b52df2ab8 I added also explicitly handled the conversion of the input representation of the jacobian via matrix ${} ^W T _I = diag({} ^W X _I, I _n)$ (with $I$ the active data representation). Unfortunately the test still fails for the `[UR10-mixed]` and `[erogcub-mixed]` cases, while it works for `[box-mixed]`. This is making me think of something related to joints that the box does not have, but I could not spot the bug here.

@diegoferigo @flferretti any idea?

File: src/jaxsim/api/frame.py [hunk 1]
```diff
@@ -257,3 +257,172 @@ def jacobian(
...
+            O_Ẋ_W = F_Ẋ_W = -F_X_W @ W_vx_WF
+            # O_J̇_WF_I = F_X_W @ (W_J̇_WL_I - W_vx_WF @ W_J_WL_I)
+            O_J̇_WF_I = jnp.zeros(shape=(6, 6 + model.dofs()))
+            O_J̇_WF_I += O_Ẋ_W @ W_J_WL_W @ T
+            O_J̇_WF_I += O_X_W @ W_J̇_WL_W @ T
+            O_J̇_WF_I += O_X_W @ W_J_WL_W @ Ṫ
+
+        case VelRepr.Mixed:
+            W_H_F = transform(model=model, data=data, frame_index=frame_index)
+            W_H_FW = W_H_F.at[0:3, 0:3].set(jnp.zeros((3, 3)))
```
Comment (diegoferigo) [inline src/jaxsim/api/frame.py:402]:
```suggestion
            W_H_FW = W_H_F.at[0:3, 0:3].set(jnp.eye(3))
```

**Review by diegoferigo — CHANGES_REQUESTED** [review]
I need some time to go through you code, but the following suggestion is definitely a bug. Let's see what happens as soon as you fix it.

File: src/jaxsim/api/frame.py (hunk 1 above)
Comment (xela-95) [reply src/jaxsim/api/frame.py]: It was that indeed!! Thank you so much, I was thinking of rocket science and it was a dummy error instead :)

Comment (diegoferigo) [reply src/jaxsim/api/frame.py]: Ahah a set of fresh eyes are often useful :eyes: I'm glad it helped, it was really a 10 seconds look :) I'm super happy that the approach with the three-derivatives components worked well also in this case.

File: src/jaxsim/api/frame.py [hunk 2]
```diff
@@ -257,3 +257,167 @@ def jacobian(
...
+            O_Ẋ_W = W_Ẋ_W = jnp.zeros((6, 6))
+            O_J̇_WF_I = jnp.zeros(shape=(6, 6 + model.dofs()))
+            O_J̇_WF_I += O_Ẋ_W @ W_J_WL_W @ T
+            O_J̇_WF_I += O_X_W @ W_J̇_WL_W @ T
+            O_J̇_WF_I += O_X_W @ W_J_WL_W @ Ṫ
+
+        case VelRepr.Body:
+            W_H_F = transform(model=model, data=data, frame_index=frame_index)
+            F_H_W = Transform.inverse(W_H_F)
+            O_X_W = F_X_W = Adjoint.from_transform(transform=F_H_W)
```
Comment (diegoferigo) [inline src/jaxsim/api/frame.py:387]:
```suggestion
            O_X_W = F_X_W = Adjoint.from_transform(transform=W_H_F, inverse=True)
```

File: src/jaxsim/api/frame.py [hunk 3]
```diff
@@ -257,3 +257,167 @@ def jacobian(
...
+                )
+                FW_v_WF = FW_J_WF_FW @ data.generalized_velocity()
+            W_v_W_FW = jnp.zeros(6).at[0:3].set(FW_v_WF[0:3])
+            W_vx_W_FW = Cross.vx(W_v_W_FW)
+            O_Ẋ_W = FW_Ẋ_W = -FW_X_W @ W_vx_W_FW
+
+            O_J̇_WF_I = jnp.zeros(shape=(6, 6 + model.dofs()))
+            O_J̇_WF_I += O_Ẋ_W @ W_J_WL_W @ T
+            O_J̇_WF_I += O_X_W @ W_J̇_WL_W @ T
+            O_J̇_WF_I += O_X_W @ W_J_WL_W @ Ṫ
```
Comment (diegoferigo) [inline src/jaxsim/api/frame.py:418]: Can we move out from the match-case these lines that are shared among all the representations?

File: tests/test_api_frame.py [hunk 4]
```diff
@@ -5,12 +5,12 @@
 
 import jaxsim.api as js
 from jaxsim import VelRepr
+from jaxsim.math.quaternion import Quaternion
 
 from . import utils_idyntree
 
 
 def test_frame_index(jaxsim_models_types: js.model.JaxSimModel):
-
```
Comment (diegoferigo) [inline tests/test_api_frame.py:13]: Nit: can you keep spaces?

File: tests/test_api_frame.py [hunk 5]
```diff
@@ -164,7 +161,132 @@ def test_frame_jacobians(
...
+    # Compute J̇.
+    O_J̇_WF_I = jax.vmap(
+        lambda frame_index: js.frame.jacobian_derivative(
+            model=model, data=data, frame_index=frame_index
+        )
+    )(frame_idxs)
+
+    assert O_J̇_WF_I.shape == (len(frame_names), 6, 6 + model.dofs())
+
+    # Compute O_J̇_WF_I using AD
```
Comment (diegoferigo) [inline tests/test_api_frame.py:217]: leftover?

**Review by diegoferigo — APPROVED** [review]
LGTM! Just minor suggestions.

File: tests/test_api_frame.py (hunk 5 above)
Comment (xela-95) [reply tests/test_api_frame.py]: Thanks, solved in https://github.com/ami-iit/jaxsim/pull/210/commits/beed30b3a584a538fe5d97f9dad30238defda4ff

File: tests/test_api_frame.py (hunk 4 above)
Comment (xela-95) [reply tests/test_api_frame.py]: Thanks, solved in https://github.com/ami-iit/jaxsim/pull/210/commits/beed30b3a584a538fe5d97f9dad30238defda4ff

File: src/jaxsim/api/frame.py (hunk 3 above)
Comment (xela-95) [reply src/jaxsim/api/frame.py]: Thanks, solved in https://github.com/ami-iit/jaxsim/pull/210/commits/beed30b3a584a538fe5d97f9dad30238defda4ff

File: src/jaxsim/api/frame.py (hunk 2 above)
Comment (xela-95) [reply src/jaxsim/api/frame.py]: Thanks, solved in https://github.com/ami-iit/jaxsim/pull/210/commits/beed30b3a584a538fe5d97f9dad30238defda4ff

**Review by flferretti — APPROVED** [review]
That's great @xela-95, thanks a lot! LGTM

**Pull request merged**